### DISCUSSION TITLE
github-actions[bot] denied permission

### DISCUSSION THREAD

**Discussion by ElvisAbner** [community/community#60392] [Actions] [answered]
### Select Topic Area

Question

### Body

I am having an issue trying to run I am the owner of the repository 'ElvisAbner/ElvisAbner' and I am encountering permission errors when the GitHub Actions workflow attempts to push changes. The error message I received is as follows: 'remote: Permission to ElvisAbner/ElvisAbner.git denied to github-actions[bot]. fatal: unable to access 'https://github.com/ElvisAbner/ElvisAbner.git/': The requested URL returned error: 403'.

I am trying to push changes to the branch 'main' of my repository. However, the github-actions[bot] user associated with the GitHub Actions workflow is being denied permission to access and push changes to the repository. As the owner of the repository, I would expect to have the necessary permissions to perform these actions.

I have verified the repository settings and my ownership status, ensuring that I have the necessary permissions to push changes. Additionally, I have reviewed the workflow configuration and confirmed that the GitHub token being used in the workflow has appropriate permissions.

I would appreciate any assistance you can provide in resolving this permission issue so that I can successfully push changes from the GitHub Actions workflow as the repository owner.

this is my code: 

` # GitHub Action for generating a contribution graph with a snake eating your contributions.

name: Generate Snake

# Controls when the action will run. This action runs every 6 hours.

on:
  schedule:
      # every 6 hours
    - cron: "0 */6 * * *"

# This command allows us to run the Action automatically from the Actions tab.
  workflow_dispatch:

# The sequence of runs in this workflow:
jobs:
  # This workflow contains a single job called "build"
  build:
    # The type of runner that the job will run on
    runs-on: ubuntu-latest

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:

    # Checks repo under $GITHUB_WORKSHOP, so your job can access it
      - uses: actions/checkout@v2

    # Generates the snake  
      - uses: Platane/snk@master
        id: snake-gif
        with:
          github_user_name: ElvisAbner
          # these next 2 lines generate the files on a branch called "output". This keeps the main branch from cluttering up.
          gif_out_path: dist/github-contribution-grid-snake.gif
          svg_out_path: dist/github-contribution-grid-snake.svg

     # show the status of the build. Makes it easier for debugging (if there's any issues).
      - run: git status

      # Push the changes
      - name: Push changes
        uses: ad-m/github-push-action@master
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          branch: main
          force: true

      - uses: crazy-max/ghaction-github-pages@v2.1.3
        with:
          # the output branch we mentioned above
          target_branch: output
          build_dir: dist
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}`

error code: remote: Permission to ElvisAbner/ElvisAbner.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/ElvisAbner/ElvisAbner.git/': The requested URL returned error: 403
Error: Invalid exit code: 128
    at ChildProcess.<anonymous> (/home/runner/work/_actions/ad-m/github-push-action/master/start.js:29:21)
    at ChildProcess.emit (node:events:527:28)
    at maybeClose (node:internal/child_process:[10](https://github.com/ElvisAbner/ElvisAbner/actions/runs/5494497480/jobs/10013323074#step:6:11)92:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5) {
  code: [12](https://github.com/ElvisAbner/ElvisAbner/actions/runs/5494497480/jobs/10013323074#step:6:13)8
}
Error: Invalid exit code: 128
    at ChildProcess.<anonymous> (/home/runner/work/_actions/ad-m/github-push-action/master/start.js:29:21)
    at ChildProcess.emit (node:events:527:28)
    at maybeClose (node:internal/child_process:1092:[16](https://github.com/ElvisAbner/ElvisAbner/actions/runs/5494497480/jobs/10013323074#step:6:17))
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)

**Comment by GhostOf0days** [comment, marked as answer]
Please go to your repo's settings, click on the 'Actions' category on the left side that is under 'Code and automation', scroll down to 'Workflow permissions', select 'Read and write permissions', click 'Allow Github Actions to create and approve pull requests', and click 'Save.'

**Reply by Husnain0092** [reply]
I tried but it didn’t work as I'm running the workflow on Repository A which is on Account 1 and want to push an entire branch code into Repository B which is on Account 2. 

It gives me the below error while push:
still same remote: Permission to Account2/RepoB.git denied to github-actions[bot].

**Reply by krunalshoppiko** [reply]
Are you found the solution?

**Reply by DSDrachmann** [reply]
This is maybe not directly related but was encountering the same issue on a project of my own where i wanted to push a change to my repo through the github action and I had made the mistake of writing this in my workflow file:

permissions:
  contents: read

Which defined my github tokens access to only read from the repo and for some reason i could not get past this with a PAT (personal access token).
After i changed it to:

permissions:
  contents: write

it was now possible to do what i wanted to do as i could now both read and write to/from the repo.

**Reply by ahmadqayyum7** [reply]
you are a life saver. i was stuck at this problem for hours and YouTube tutorials were just making me angry as they were meaningless. again, a life saver!!!!!!

**Reply by iscpatricio92** [reply]
thanks for solution 👏🏻

**Reply by jvxiao** [reply]
> This is maybe not directly related but was encountering the same issue on a project of my own where i wanted to push a change to my repo through the github action and I had made the mistake of writing this in my workflow file:
> 
> permissions: contents: read
> 
> Which defined my github tokens access to only read from the repo and for some reason i could not get past this with a PAT (personal access token). After i changed it to:
> 
> permissions: contents: write
> 
> it was now possible to do what i wanted to do as i could now both read and write to/from the repo.

My god, I'm stuck for the same reason.
